components/org.wso2.carbon.identity.captcha/src/main/java/org/wso2/carbon/identity/captcha/validator/FailLoginAttemptValidator.java

```diff
@@ -21,48 +21,51 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.wso2.carbon.identity.application.authentication.framework.AuthenticationDataPublisher;
 import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
-import org.wso2.carbon.identity.application.authentication.framework.context.SessionContext;
 import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkConstants;
 import org.wso2.carbon.identity.application.common.model.User;
 import org.wso2.carbon.identity.captcha.exception.CaptchaException;
 import org.wso2.carbon.identity.captcha.util.CaptchaConstants;
 import org.wso2.carbon.identity.captcha.util.CaptchaUtil;
 import org.wso2.carbon.identity.core.bean.context.MessageContext;
 import org.wso2.carbon.identity.core.handler.AbstractIdentityMessageHandler;
+import org.wso2.carbon.identity.core.model.IdentityEventListenerConfig;
 import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventConstants.EventName;
+import org.wso2.carbon.identity.event.IdentityEventConstants.EventProperty;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.event.event.Event;
+import org.wso2.carbon.identity.event.handler.AbstractEventHandler;
 import org.wso2.carbon.user.core.util.UserCoreUtil;
 
-import javax.servlet.http.HttpServletRequest;
 import java.util.Map;
 
 /**
  * FailLoginAttemptValidator.
  */
-public class FailLoginAttemptValidator extends AbstractIdentityMessageHandler implements AuthenticationDataPublisher {
+public class FailLoginAttemptValidator extends AbstractEventHandler {
 
     private static Log log = LogFactory.getLog(FailLoginAttemptValidator.class);
 
     @Override
     public String getName() {
-        return "FailLoginAttemptValidator";
+        return "failLoginAttemptValidator";
     }
 
-    @Override
-    public boolean canHandle(MessageContext messageContext) {
-        return true;
-    }
 
     @Override
-    public void publishAuthenticationStepSuccess(HttpServletRequest httpServletRequest,
-                                                 AuthenticationContext authenticationContext, Map<String, Object> map) {
-
+    public void handleEvent(Event event) throws IdentityEventException {
+        AuthenticationContext context = (AuthenticationContext) event.getEventProperties().get(EventProperty.CONTEXT);
```

Comment:
It's always a good practice to check whether this is an instance of , before casting